config/base/derive/src/config.rs

```diff
@@ -0,0 +1,477 @@
+    fn parse(input: ParseStream) -> syn::Result<Self> {
+        Ok(Self {
+            _ident: utils::parse_const_ident(input, attrs::ENV_PREFIX)?,
+            _eq: input.parse()?,
+            prefix: input.parse()?,
+        })
+    }
+}
+
+pub(crate) struct Inner {
+    _ident: Ident,
+}
+
+impl Parse for Inner {
+    fn parse(input: ParseStream) -> syn::Result<Self> {
+        Ok(Self {
+            _ident: utils::parse_const_ident(input, attrs::INNER)?,
+        })
+    }
+}
+
+pub(crate) struct SerdeAsStr {
+    _ident: Ident,
+}
+
+impl Parse for SerdeAsStr {
+    fn parse(input: ParseStream) -> syn::Result<Self> {
+        Ok(Self {
+            _ident: utils::parse_const_ident(input, attrs::SERDE_AS_STR)?,
+        })
+    }
+}
+
+pub(crate) fn impl_load_env(
+    field_idents: &[&Ident],
+    inner: &[bool],
+    option: &[bool],
+    lvalue: &[proc_macro2::TokenStream],
+    as_str: &[bool],
+    field_ty: &[Type],
+    field_environment: &[String],
+) -> proc_macro2::TokenStream {
+    let set_field = field_ty
+        .iter()
+        .zip(field_idents.iter())
+        .zip(as_str.iter())
+        .zip(lvalue.iter())
+        .zip(option.iter())
+        .map(|((((ty, ident), &as_str_attr), l_value), &is_option)| {
+            let is_string = if let Type::Path(TypePath { path, .. }) = ty {
+                path.is_ident("String")
+            } else {
+                false
+            };
+            // println!("Type in ")
+            // TODO: no need for separate fn
+            let set_field = if is_string {
+                quote! { #l_value = var }
+            } else if as_str_attr {
+                quote! {
+                    #l_value = serde_json::from_value(var.into())
+                        .map_err(|e| iroha_config_base::derive::Error::field_error(stringify!(#ident), e))?
+                }
+            } else if is_option {
+                quote! {
+                    #l_value = Some(serde_json::from_value(var.into())
+                        .map_err(|e| iroha_config_base::derive::Error::field_error(stringify!(#ident), e))?)
+                }
+            }
+            else {
+                quote! {
+                    #l_value = serde_json::from_str(&var)
+                        .map_err(|e| iroha_config_base::derive::Error::field_error(stringify!(#ident), e))?
+                }
+            };
+            (set_field, l_value)
+        })
+        .zip(field_environment.iter())
+        .zip(inner.iter())
+        .zip(option.iter())
+        .map(|((((set_field, l_value), field_env), &inner_thing), &is_option)| {
+            let inner_thing2 = if inner_thing && is_option {
+                quote! {
+                    // <#inner>::load_environment()?;
+                    #l_value.unwrap().load_environment()?;
+                    // if let Ok(config) = <#inner>::load_environment() {
+                    //     #l_value = Some(config)
+                    // };
+                }
```

Comment:
An unfinished mess here as the whole `Configurable` trait is not implemented yet for proxy and I wasn't able to test this piece. What gets done to load the inner configuration of fields wrapped inside `Option` still needs to be thought through.